vita3k/kernel/src/thread/thread.cpp

```diff
@@ -117,16 +117,18 @@ SceUID create_thread(Ptr<const void> entry_point, KernelState &kernel, MemState
 
     alloc_name = fmt::format("TLS for thread {} (#{})", name, thid);
 
-    constexpr auto tls_size = 0x800;
-    auto tls_address_ptr = Ptr<void>(alloc(mem, tls_size, alloc_name.c_str()));
-    auto tls_address = tls_address_ptr.address() + tls_size;
+    auto tls_size = kernel_tls_size + kernel.tls_msize;
```

Comment:
What happens if you just use tls_msize when its not zero?

Reply:
Maybe it's the same tls, but I think it's unlikely.

Reply:
Oh cool that sounds compelling. I want to check out more about this but I have a feeling youve gotten there.